### PR TITLE
2060101: [1.24] Print correct status, when access mode has changed

### PR DESCRIPTION
**Pull request by jirihnidek**
* Backport two PRs to 1.24 branch. Original PRs:
  * https://github.com/candlepin/subscription-manager/pull/3049
    * Required some changes to make code Python 2 compatible
  * Part of this PR: https://github.com/candlepin/subscription-manager/pull/2321
* Card ID: ENT-5327

### CODE REVIEW

**Comment by ptoscano** [comment]
Please rebase on top of the current 1.24 branch, so it gets the recent GitHub Action additions.

**Comment by jirihnidek** [comment]
Hi @pinotree, FYI it is rebased

File: src/subscription_manager/cert_sorter.py [hunk 1]
```diff
@@ -259,8 +260,33 @@ def _scan_entitlement_certs(self):
...
+            DISABLED: _("Disabled"),
+            UNKNOWN: _("Unknown")
+        }
+        return status_map
+
     def get_system_status(self):
-        return STATUS_MAP.get(self.system_status, STATUS_MAP['unknown'])
+        """
+        Return string with translated status of system
+        """
```
Comment (ptoscano) [inline src/subscription_manager/cert_sorter.py:287]: while adding comments is generally a good thing, this was not in the original commit

**Review by ptoscano — CHANGES_REQUESTED** [review]
Sorry for the late answer, testing things on el7 took me more time; it is generally OK, there are few things to tweak (I added inline comments); one more thing I cannot add an inline comment is:
In `ComplianceManager.load()` there is:
```python
        self.system_status = 'unknown'
```
As part of the changes of the backport of commit ad23dcf1fdab56fb03fae895a8d86d8f81c3a3da, this needs to be changed to
```python
        self.system_status = UNKNOWN
```

Thanks!

**Comment by jirihnidek** [comment]
> ```python
> self.system_status = 'unknown'
> ```

Good catch! Thanks. I also noticed that `get_status_for_icon()` used strings ("invalid" and "partial").

**Review by ptoscano — APPROVED** [review]

**Pull request merged**